core/pkg/launch/job_builder.go

```diff
@@ -734,15 +735,23 @@ func (j *JobBuilder) HandleUseArtifactRecord(record *service.Record) {
 // Makes job input schema into a json string to be stored as artifact metdata.
 func (j *JobBuilder) makeJobMetadata(output *data_types.TypeRepresentation) (string, error) {
 	metadata := make(map[string]interface{})
+	input_types := make(map[string]interface{})
+	if len(j.configFiles) > 0 {
+		files := make(map[string]interface{})
+		for _, configFile := range j.configFiles {
+			files[configFile.relpath] = j.generateConfigFileSchema(configFile)
+		}
+		input_types["files"] = files
+	}
```

Comment:
it feels like this logic could live closer to the config file handling?

Reply:
It could for config files specified by the user but not for the `wandb.config`. Handling the config files here allows us to use the same library in the internal process for filtering the tree and then extracting a schema from the run config and any user-provided config files. If I move this to the user process I need to rewrite all of this tree filtering business in python and keep those implementations in sync.